src/controllers/match2/index.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  ValidateError,
+  Controller,
+  Post,
+  Get,
+  Route,
+  Response,
+  Body,
+  SuccessResponse,
+  Tags,
+  Security,
+  Path,
+} from 'tsoa'
+import type { Logger } from 'pino'
+
+import { logger } from '../../lib/logger'
+import Database from '../../lib/db'
+import { BadRequest, NotFound } from '../../lib/error-handler/index'
+import { getMemberByAddress, getMemberBySelf } from '../../lib/services/identity'
+import { Match2Request, Match2Response, Match2State } from '../../models/match2'
+import { UUID } from '../../models/uuid'
+
+@Route('match2')
+@Tags('match2')
+@Security('bearerAuth')
+export class Match2Controller extends Controller {
+  log: Logger
+  db: Database
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/match2' })
+    this.db = new Database()
+  }
+
+  /**
+   * A Member proposes a new match2 for an order and a capacity by referencing each demand.
+   * @summary Propose a new match2
+   */
+  @Post()
+  @Response<BadRequest>(400, 'Request was invalid')
+  @Response<ValidateError>(422, 'Validation Failed')
+  @SuccessResponse('201')
+  public async proposeMatch2(
+    @Body() { demandA: demandAId, demandB: demandBId }: Match2Request
+  ): Promise<Match2Response> {
+    const [demandA] = await this.db.getDemand(demandAId)
+    if (!demandA) {
+      throw new BadRequest('Demand A not found')
+    }
+
+    const [demandB] = await this.db.getDemand(demandBId)
+    if (!demandB) {
+      throw new BadRequest('Demand B not found')
+    }
+
+    if (demandA.subtype === demandB.subtype) {
```

Comment:
thanks, missed that on the process flow, I thought it could be either way round. Added a stricter check and updated tests https://github.com/digicatapult/dscp-matchmaker-api/pull/31/commits/61b4c7104eff4cf017108351e771d2ee88bb9da3